src/helper/client.cpp

```diff
@@ -73,33 +74,53 @@ bool handle_message(client &client, const network::base_broker &broker,
     }
 
     bool send_error = false;
+    bool result = true;
     try {
         auto msg = broker.recv(initial_timeout);
         return maybe_exec_cmd_M<Ms...>(client, msg);
     } catch (const client_disconnect &) {
         SPDLOG_INFO("Client has disconnected");
+        // When this exception has been received, we should stop hadling this
+        // particular client.
+        result = false;
+    } catch (const std::out_of_range &e) {
+        // The message received was too large, in theory this should've been
+        // flushed and we can continue handling messages from this client,
+        // however we need to report an error to ensure the client is in a good
+        // state.
+        SPDLOG_WARN("Failed to handle message: {}", e.what());
```

Comment:
Will we be able to know what error happened based on the `e.what()`? I can see almost all catch section throw to the log the same start of the message `Failed to handle message:`